lib/devices/Roborock.js

```diff
@@ -505,33 +505,40 @@ class Roborock extends MiioVacuum {
         return await this.sendLocal("app_goto_target", [x_coord, 51200 - y_coord], {});
     }
 
-    async startCleaningZoneByName(zones) {
-        let areas = this.configuration.get("areas");
-        zones = Array.isArray(zones) ? zones : [ zones ];
-        let zonesToClean = [];
+    /** @param {Array<number>} zone_ids */
+    async startCleaningZonesById(zone_ids) {
+        let zones = this.configuration.getZones();
+        /** @type {Array<import('../Configuration').Area>} */
+        let areasToClean = [];
+        /** @type {Array<string>} */
         let zoneNames = [];
-        zones.forEach(function (zone_id) {
-            let area = areas.find(e => Array.isArray(e) && e[0] === zone_id);
-            if (area && Array.isArray(area[1])) {
-                area[1].forEach(function (zone) {
-                    zonesToClean.push(zone);
-                });
-                zoneNames.push(area[0]);
+        zone_ids.forEach((zone_id) => {
+            let zone = zones.get(zone_id);
+            if (zone) {
+                areasToClean.concat(zone.areas);
```

Comment:
Array.concat() returns the new array and not modifies the existing arrays.
areasToClean = areasToClean.concat(zone.areas);

Reply:
Thanks for catching!